portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,9 +55,32 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("comments-input");
-    commentsStorage.add(comment);
+    max_comments_num = getCommentsMaxNum(request);
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("contents", comment);
+    datastore.put(commentEntity);
     response.setContentType("text/html;");
     response.getWriter().println("Thank you for your comments!");
-    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * Returns the maximum number of comments entered by the viewer, or the default number of comments
+   * (3) if the choice was invalid.
+   */
+  private int getCommentsMaxNum(HttpServletRequest request) {
+    String maxCommentsNumString = request.getParameter("max-num-comments");
+    int defaultNum = 3;
+    int maxCommentsNum;
+    try {
+      maxCommentsNum = Integer.parseInt(maxCommentsNumString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + maxCommentsNumString);
```

Comment:
LOG

Reply:
It has now been resolved by avoiding writing this extra function to get the inserted number.